src/box/space.c

```diff
@@ -383,6 +384,14 @@ space_before_replace(struct space *space, struct txn *txn,
 
 	if (index_get(index, key, part_count, &old_tuple) != 0)
 		return -1;
+	if (old_tuple != NULL) {
+		if (tuple_is_compressed(old_tuple)) {
+			old_tuple = tt_decompress_tuple_new(space, old_tuple);
+			if (old_tuple == NULL)
+				return -1;
+		}
+		tuple_ref(old_tuple);
+	}
```

Comment:
This would better be done somewhere in `index_get` IMO.

Reply:
In many cases `index_get` should return tuple as is (without decompression), so i think it's better to do decompression here

Reply:
All these cases are inside memtx AFAIU. You can introduce a new function for them.